docs/sources/flow/reference/stdlib/json_path.md

```diff
@@ -0,0 +1,36 @@
+---
+aliases:
+- ../../configuration-language/standard-library/json_path/
+title: json_path
+---
+
+# json_path
+
+The `json_path` function lookup values using [jsonpath](https://goessner.net/articles/JsonPath/) syntax.
+
+The function expects 2 strings. First string is the json string to lookup values from. Second string is the jsonpath expression.
+
+`json_path` always return a list of values. If the jsonpath expression does not match any values, an empty list is returned.
+
+> Remember to escape double quotes when passing JSON string literals to `json_path`.
+>
+> For example, the JSON value `{"key": "value"}` is properly represented by the
+> string `"{\"key\": \"value\"}"`.
+
+## Examples
+
+```
+> json_path("{\"key\": \"value\"}", ".key")
+["value"]
+
+
+> json_path("[{\"name\": \"Department\",\"value\": \"IT\"},{\"name\":\"TestStatus\",\"value\":\"Pending\"}]", "[?(@.name == \"Department\")].value")
+["IT"]
+
+> json_path("{\"key\": \"value\"}", ".nonexists")
+[]
+
+> json_path("{\"key\": \"value\"}", ".key")[0]
+value
+
+```
```

Comment:
As an aside: 

Wow, River really needs alternative ways of defining strings so users don't have to escape double quotes like this constantly.

Reply:
I would recomend backticks here. Backticks are already used in other languages like go or js.

Reply:
Backticks (Go, JS) or single quotes (YAML) seem like fine options. 

Though, backticks in JS indicate an interpolated string, which I currently proposed a different syntax for in #2972. Maybe #2972 should change its recommendation to backticks?

Reply:
hm.

Is there any downside by keeping the string interpolation for backticks + recommend them to avoid quote escaping?

In JS, the developers are using backtick to avoid escaping, too. (https://stackoverflow.com/questions/53180063/how-do-i-save-a-json-string-in-javascript-file). But JS developers are always a bit crazy.

An alternative would be textblock, like:

```python
i = """
{"hello": "world"}
"""
```

I know this from syntax from java, python and markdown (using 3 backticks)

But I personally like the backtick thing a bit more.

Reply:
In my prototype, I found that there's issues in general with using `"` anywhere in the interpolation character (`"` or `"""`), since the formatter code needs to recognize when it needs to escape inner `"` or not.

For that reason alone, to keep the syntax cleaner, using a backtick for interpolation might make sense, with support for escaping interpolation and escaping backticks with `\`. 

However, Go's version of backticks is much more literal, where you can't even escape backticks themselves. I guess that would be the downside of using backticks for string interpolation and avoiding double-quotes: it would mean that there is still no string character which gives you string literals without the need to escape anything (but maybe that's fine).

Reply:
+1 backticks for me

Reply:
(We should probably move this discussion to #2972 but it does sound like backticks are probably a good choice for implementation simplicity, familiarity, and uniqueness to not need frequent escaping.)